adapters/bridge.go

```diff
@@ -19,38 +19,68 @@ type Bridge struct {
 
 // Perform sends a POST request containing the JSON of the input RunResult to
 // the external adapter specified in the BridgeType.
+// It records the RunResult reutrned to it, and optionally marks the RunResult pending.
+//
+// If the Perform is resumed with a pending RunResult, the RunResult is marked
+// not pending and the RunResult is returned.
 func (ba *Bridge) Perform(input models.RunResult, _ *store.Store) models.RunResult {
-	in, err := json.Marshal(&input.Output)
+	if input.Pending {
+		return markNotPending(input)
```

Comment:
In this instance, don't we want to save the payload received from resuming to the run result?

Reply:
The payload received from resuming is now the `input`, so it is saved by the job runner once it is returned.